packages/runtime-core/src/helpers/renderList.ts

```diff
@@ -32,5 +32,5 @@ export function renderList(
       }
     }
   }
```

Comment:
I'd instead add an `else` branch here and in addition add a warning for invalid `v-for` value.

Reply:
I don't know if a `warn` is necessary, because it is already called once, for undefined properties:

<details><summary>Show errors</summary>

![image](https://user-images.githubusercontent.com/32134586/69682869-a69f9500-1091-11ea-93e9-866da4a0cb54.png)

</details>

---

If you can answer, what's the trade-off of not assign the `ret` at the top level?

Reply:
Yeah I just checked and we don't emit warnings for invalid values in v2.
Setting the ret at top level means you always allocate an extra empty array for valid values.